lms/djangoapps/discussion_api/tests/test_views.py

```diff
@@ -866,6 +896,31 @@ def test_pagination(self):
             }
         )
 
+    @ddt.data(
+        (True, "endorsed_comment"),
+        ("true", "endorsed_comment"),
+        ("1", "endorsed_comment"),
+        (False, "non_endorsed_comment"),
+        ("false", "non_endorsed_comment"),
+        ("0", "non_endorsed_comment"),
+    )
```

Comment:
That sounds like a good idea.  Rather than doing explicit CHOICES, we could do a true case-insensitive comparision (input.lower() == 'true').  The result would either be True, False, None (on an empty string or None), or an error (Which should reach the user as an HTTP 400).